pilot/pkg/controllers/untaint/nodeuntainter.go

```diff
@@ -96,12 +96,12 @@ func (n *NodeUntainter) setup(stop <-chan struct{}) {
 			return nil
 		}
 		log.Debugf("pod %s on node %s ready!", p.Name, p.Spec.NodeName)
-		return p
+		return &p
 	}, krt.WithStop(stop))
 
 	// these are all the nodes that have a ready cni pod. if the cni pod is ready,
 	// it means we are ok scheduling pods to it.
-	readyCniNodes := krt.NewCollection(readyCniPods, func(ctx krt.HandlerContext, p v1.Pod) *v1.Node {
+	readyCniNodes := krt.NewCollection(readyCniPods, func(ctx krt.HandlerContext, p *v1.Pod) **v1.Node {
```

Comment:
Personally i donot like this pointer to a pointer type, it makes it hard to maintain and easy to panic

Reply:
I'm not sure why 2 pointers makes it more likely to panic? should be about the same usage. 

The alternative is to make it return (T, bool) which tends to be quite tedious

Reply:
btw at no point is anything really interacting with these double pointers - when they use the collection it's a single pointer. it's only in the return value for the most part

Reply:
hmm, the generic support makes it quite hard to understand the i/o type, maybe i miss the ctx. Will come back after more careful look

Reply:
The implement is right, but some other `NewCollection` caller, we donot use double pointer. What is the consideration
The way to remove double pointer is 
replace `TransformationSingle[I, O any] func(ctx HandlerContext, i I) *O` with 
`TransformationSingle[I, O any] func(ctx HandlerContext, i I) O`

Reply:
that would mean it's impossible for it to not generate something for an input. Really what we want the output to he is `Option<T>` but not really feasible in go (technically we could but I don't think we should)

Reply:
The consideration is the type of the collection needs to be valid - implementing GetKey etc. For k8s and protobuf types, these types are expected to be carried around as pointers.